### PR TITLE
Add examples to demo IndexOutOfBoundsException while using Collectins…

### PR DESCRIPTION
**Pull request by vishal1023**
Add examples to demo IndexOutOfBoundsException while using the Collectins copy method and other working demos to copy elements from one list to another list

### CODE REVIEW

File: core-java-modules/core-java-exceptions-3/src/test/java/com/baeldung/CopyListUsingAddAllMethodDemoUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+import static com.baeldung.CopyListUsingAddAllMethodDemo.copyList;
+import static org.junit.jupiter.api.Assertions.*;
+
+class CopyListUsingAddAllMethodDemoUnitTest {
+
+    @Test
+    void shouldMakeACopyOfSourceList() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-exceptions-3/src/test/java/com/baeldung/CopyListUsingAddAllMethodDemoUnitTest.java:14]: Make the test method names follow the givenX_whenY_thenZ convention (givenX is optional).

**Review by KevinGilmore — COMMENTED** [review]
update test method names as per convention

File: core-java-modules/core-java-exceptions-3/src/main/java/com/baeldung/CopyListUsingConstructorDemo.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.baeldung;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CopyListUsingConstructorDemo {
+
+    static List<Integer> copyList(List<Integer> source) {
+
```
[the remark this answers is not in the document]
Comment (vishal1023) [reply core-java-modules/core-java-exceptions-3/src/main/java/com/baeldung/CopyListUsingConstructorDemo.java]: removed the blank lines

File: core-java-modules/core-java-exceptions-3/src/test/java/com/baeldung/CopyListUsingAddAllMethodDemoUnitTest.java (hunk 1 above)
Comment (vishal1023) [reply core-java-modules/core-java-exceptions-3/src/test/java/com/baeldung/CopyListUsingAddAllMethodDemoUnitTest.java]: updated the test method names as per convention

**Comment by vishal1023** [comment]
creating a different pull request for this with a specific branch.

**Pull request closed without merging**